wandb/apis/public.py

```diff
@@ -1024,6 +1024,40 @@ def update(self):
         )
         self.summary.update()
 
+    def delete(self, delete_artifacts=False):
+        """
+        Deletes the given run from the wandb backend.
+        """
+        mutation = gql(
+            """
+            mutation DeleteRun(
+                $id: ID!,
+                %s
+            ) {
+                deleteRun(input: {
+                    id: $id,
+                    %s
+                }) {
+                    clientMutationId
+                }
+            }
+        """ %
+            # Older backends might not support the 'deleteArtifacts' argument,
+            # so only supply it when it is explicitly set.
+            (
+                "$deleteArtifacts: Boolean" if delete_artifacts else "",
+                "deleteArtifacts: $deleteArtifacts" if delete_artifacts else ""
+            )
+        )
+
+        return self.client.execute(
+            mutation,
+            variable_values={
+                "id": self.storage_id,
+                "deleteArtifacts": delete_artifacts,
```

Comment:
shouldnt we not pass this deleteArtifacts if delete_artifacts is False?

Reply:
The extraneous variables are disregarded, so it doesn't cause an issue. I felt it made the code simpler, but I can see it being confusing as well.

Reply:
i like cleaner code,  i didnt know that was the behavior.  👍